### PR TITLE
Allow testing the API itself

### PR DESCRIPTION
**Pull request by j-delaney**

### CODE REVIEW

**Comment by j-delaney** [comment]
For this to work on Travis `go build` would have to be called first.

File: redoctober_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+	"github.com/cloudflare/redoctober/core"
+)
+
+const baseURL = "https://localhost:8080/"
+
+func init() {
+	// If these tests have been run previously and failed it's possible there is an existing
+	// disk record we need to remove.
+	os.Remove("diskrecord_test.json")
```
Comment (grittygrease) [inline redoctober_test.go]: Alternatively, you could use the in-memory vault.

**Comment by grittygrease** [comment]
Looks like Travis is failing.

**Comment by kisom** [comment]
```
$ go test github.com/cloudflare/redoctober...

SIGQUIT: quit

PC=0x478eb8
```

Go's testing library will send a SIGQUIT to the process before SIGKILL. This means the test is running too long and getting killed.

**Comment by j-delaney** [comment]
PR is finished and ready for review now.

**Comment by grittygrease** [comment]
@jkroll-cf @kisom, do you have any comments on this PR?

File: redoctober_test.go [hunk 2]
```diff
@@ -0,0 +1,684 @@
...
+		return
+	}
+
+	buffer := bytes.NewBuffer(jsonBytes)
+	response, err = http.Post(baseURL+api, "text/json", buffer)
+	if err != nil {
+		return
+	}
+
+	respBytes, err = ioutil.ReadAll(response.Body)
```
Comment (kisom) [inline redoctober_test.go]: You should close the response body after reading it; you can do this before checking the error; alternatively, defer the response body close after posting.

File: redoctober_test.go [hunk 3]
```diff
@@ -0,0 +1,684 @@
...
+	if _, _, err := post("create-user", createUserInput2); err != nil {
+		teardown(t, cmd)
+		t.Fatalf("Error creating user 2, %v", err)
+	}
+	if _, _, err := post("create-user", createUserInput3); err != nil {
+		teardown(t, cmd)
+		t.Fatalf("Error creating user 3, %v", err)
+	}
+
+	// Encrypt some data.
```
Comment (kisom) [inline redoctober_test.go]: The comment here is probably extraneous.

**Comment by kisom** [comment]
Once the post body closing is fixed, LGTM.

**Comment by j-delaney** [comment]
@kisom Fixed post body closing and squashed commits

**Comment by kisom** [comment]
LGTM.

**Pull request merged**